src/commands/deployments/disconnectRepo.ts

```diff
@@ -3,13 +3,21 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { DeploymentsTreeItem } from "vscode-azureappservice";
+import { DeploymentsTreeItem, disconnectRepo as disconnectRepository } from "vscode-azureappservice";
 import { IActionContext } from "vscode-azureextensionui";
 import { ext } from "../../extensionVariables";
+import { localize } from '../../localize';
+import { SlotTreeItemBase } from '../../tree/SlotTreeItemBase';
 
 export async function disconnectRepo(context: IActionContext, node?: DeploymentsTreeItem): Promise<void> {
     if (!node) {
         node = await ext.tree.showTreeItemPicker<DeploymentsTreeItem>(DeploymentsTreeItem.contextValueConnected, context);
     }
-    await node.disconnectRepo(context);
+
+    if (node.parent instanceof SlotTreeItemBase) {
+        await disconnectRepository(context, node.parent.client, node.parent.root);
+        await node.refresh();
+    } else {
+        throw Error(localize('actionNotSupported', 'Action not supported'));
```

Comment:
Is this an appropriate error message?

Reply:
Should this ever be possible? If not, make sure to add "Internal error" like [here](https://github.com/microsoft/vscode-azurefunctions/blob/master/src/utils/nonNull.ts#L24)

We have a telemetry query setup specifically for those errors. Since there's nothing the user could do, it's more important that we can find this in telemetry than for them to understand it